examples/stackoverflow/index.js

```diff
@@ -0,0 +1,123 @@
+const express = require('express');
+const superagent = require('superagent');
+const { v4: uuidv4 } = require('uuid');
+const cookieSession = require('cookie-session');
+
+const everyauth = require('@fusebit/everyauth-express');
+
+const app = express();
+const port = 3000;
+
+// Get userId from the authorization redirect or via session if already authorized.
+const validateSession = (req, res, next) => {
+  if (req.query.userId) {
+    req.session.userId = req.query.userId;
+  }
+  if (!req.session.userId) {
+    return res.redirect('/');
+  }
+  return next();
+};
+
+app.use(
+  cookieSession({
+    name: 'session',
+    secret: 'secret',
+  })
+);
+
+app.set('view engine', 'pug');
+
+app.get('/', (req, res) => {
+  if (!req.session.userId) {
+    return res.redirect(`/authorize/${uuidv4()}`);
+  }
+  res.redirect('/finished');
+});
+
+app.use(
+  '/authorize/:userId',
+  everyauth.authorize('stackoverflow', {
+    // The endpoint of your app where control will be returned afterwards
+    finishedUrl: '/finished',
+    // The user ID of the authenticated user the credentials will be associated with
+    mapToUserId: (req) => req.params.userId,
+  })
+);
+
+const stackOverflowApi = ({ access_token, client_key }) => {
+  return async (path, extraParams = {}) => {
+    const url = new URL(`https://api.stackexchange.com/2.3/${path}`);
+    url.searchParams.append('key', client_key);
+    url.searchParams.append('access_token', access_token);
+    url.searchParams.append('site', 'stackoverflow');
+    Object.keys(extraParams).forEach((key) => url.searchParams.append(key, extraParams[key]));
+    return await superagent.get(url.toString());
+  };
+};
+
+app.get('/finished', validateSession, async (req, res) => {
+  const userCredentials = await everyauth.getIdentity('stackoverflow', req.session.userId);
+  const { client_key, access_token } = userCredentials.native;
+  const stackOverflowRequest = stackOverflowApi({ access_token, client_key });
+  const meResponse = await stackOverflowRequest('me');
+  const userInfo = JSON.parse(meResponse.text);
+  const user = userInfo.items[0];
+  const currentUserAnswers = await stackOverflowRequest(`users/${user.user_id}/answers`, {
```

Comment:
Be more generous with whitespace.  Add blank lines between contextual blocks - this one for getting the credential, that one for setting up the API, etc. - so that the code is more clear and cleanly delineated.  Happy to help share some techniques over zoom.

Reply:
Organized the code, let me know how it looks now.